### PR TITLE
[Metrics API/SDK] Change Meter API/SDK to return nostd::unique_ptr for Sync Instruments

### LINKED ISSUE #1700: [Metrics API/SDK] Change Meter API/SDK to return nostd::unique_ptr for Sync Instruments
The Sync Instruments created through Meter API are not used internally within SDK. So it would be good to return nostd::unique_ptr for them. Changes are required in API and SDK for
 - Counter ( UINT64 and Double)
 - Histogram ( UINT64 and Double)
 - UpDownCounter ( INT64 and Double)

The Async Instruments can continue to return `nostd::shared_ptr`.
As part of this or separate PR, we can also maintain the shared_ptr of Async Instrument (instead of current raw pointer) in Observable Registry https://github.com/open-telemetry/opentelemetry-cpp/blob/ac105bf419ba651adbe25552c67fb83b2a7dc80f/sdk/include/opentelemetry/sdk/metrics/state/observable_registry.h#L24

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1700 (issue)

## Changes

The Sync Instruments created through Meter API return `nostd::unique_ptr`.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by owent — APPROVED** [review]
LGTM

File: sdk/include/opentelemetry/sdk/metrics/state/observable_registry.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #pragma once
+#include <future>
```
Comment (lalitb) [inline sdk/include/opentelemetry/sdk/metrics/state/observable_registry.h:5]: Nit - Can we move this header within enable macro if it is required.

**Review by lalitb — APPROVED** [review]
LGTM. Thanks

File: sdk/include/opentelemetry/sdk/metrics/state/observable_registry.h (hunk 1 above)
Comment (esigo) [reply sdk/include/opentelemetry/sdk/metrics/state/observable_registry.h]: thanks, cleaned

**Comment by esigo** [comment]
waiting to merge #1686 first.

**Review by marcalff — APPROVED** [review]
LGTM

**Pull request merged**